src/main/groovy/de/dkfz/execution/jobs/Job.groovy

```diff
@@ -0,0 +1,359 @@
+ */
+
+package de.dkfz.execution.jobs
+
+import de.dkfz.config.AppConfig
+
+import de.dkfz.config.ExecutionService
+import de.dkfz.eilslabs.tools.logging.IOHelperMethods
+import de.dkfz.eilslabs.tools.logging.LoggerWrapper
+
+import java.util.concurrent.atomic.AtomicLong
+
+
+@groovy.transform.CompileStatic
+public class Job {
+
+    public static final String ERR_MSG_ONLY_ONE_JOB_ALLOWED = "A job object is not allowed to run several times.";
+    public static final String ENV_LINESEPARATOR = System.getProperty("line.separator");
+
+    private static final LoggerWrapper logger = LoggerWrapper.getLogger(Job.class.getSimpleName());
+
+    private JobType jobType = JobType.STANDARD;
+
+    private final List<Job> arrayChildJobs = new LinkedList<>();
+
+    private static AtomicLong absoluteJobCreationCounter = new AtomicLong();
+
+    /**
+     * The name of the job which should be passed to the execution system.
+     */
+    public final String jobName;
+
+    /**
+     * An internal job creation count. Has nothing to do with e.g. PBS / cluster / process id's!
+     */
+    public final long jobCreationCounter = absoluteJobCreationCounter.incrementAndGet();
+
+    /**
+     * The tool you want to call.
+     */
+    public String toolID;
+
+    //public String toolMD5;
+
+    private boolean isDirty;
+
+    /**
+     * Parameters for the tool you want to call
+     */
+    private final Map<String, String> parameters;
+
+    /**
+     * Keeps a list of all unchanged, initial parameters, including default job parameters.
+     */
+    private final Map<String, Object> allRawInputParameters;
+    /**
+     * If you want to generated arrays use this. <p>You can do things like: n-m,
+     */
+    public final List<String> arrayIndices;
+    /**
+     * Provide a list of files if you want to generate job dependencies.
+     */
+    public transient final List<Job> parentFiles;
+
+    private Map<String, Object> initialInputParameters = [:]
+    /**
+     * You should provide i.e. job ids of qsub jobs to automatically create job
+     * dependencies.
+     */
+    public final List<JobDependencyID> dependencyIDs = new LinkedList<JobDependencyID>();
+
+    /**
+     * Temporary value which defines the jobs jobState.
+     */
+    protected transient JobState currentJobState;
+
+    private List<ProcessingCommands> processingCommand = new LinkedList<ProcessingCommands>();
+
+    /**
+     * Command object of last execution.
+     */
+    private transient Command lastCommand;
+
+    /**
+     * Stores the result when the job was executed.
+     */
+    private JobResult runResult;
+
+    /**
+     *
+     * @param jobName
+     * @param defaultParameters
+     * @param arrayIndices
+     * @param inputParameters
+     * @param parentFiles
+     */
+    public Job(String jobName, Map<String, String> defaultParameters, List<String> arrayIndices, Map<String, Object> inputParameters, List<Job> parentFiles) {
```

Comment:
default parameters: pid, analysis directory, script/tool id
inputparameters: files, strings. quite a lot of it. anything which is configured.